File: src/main/java/no/novari/flyt/history/repository/projections/InstanceFlowSummaryNativeProjection.java
package no.novari.flyt.history.repository.projections;

import java.time.Instant;

public interface InstanceFlowSummaryNativeProjection {
    Long getSourceApplicationId();

    String getSourceApplicationIntegrationId();

    String getSourceApplicationInstanceId();

    Long getIntegrationId();

    Long getLatestInstanceId();

    Instant getLatestUpdate();

    String getLatestStatusEventName();

    String getLatestStorageStatusEventName();

    String getDestinationInstanceIds();
}

File: src/main/kotlin/no/novari/flyt/history/repository/projections/InstanceFlowSummaryNativeProjection.kt
package no.novari.flyt.history.repository.projections

import java.time.Instant

interface InstanceFlowSummaryNativeProjection {
    fun getSourceApplicationId(): Long?

    fun getSourceApplicationIntegrationId(): String?

    fun getSourceApplicationInstanceId(): String?

    fun getIntegrationId(): Long?

    fun getLatestInstanceId(): Long?

    fun getLatestUpdate(): Instant?

    fun getLatestStatusEventName(): String?

    fun getLatestStorageStatusEventName(): String?

    fun getDestinationInstanceIds(): String?
}
